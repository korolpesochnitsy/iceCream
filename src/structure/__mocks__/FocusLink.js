import React from 'react';

const FocusLink = ({ to, children, ...props }) => (
  <a href={to} {...props}>
    {children}
  </a>
);

export default FocusLink;
