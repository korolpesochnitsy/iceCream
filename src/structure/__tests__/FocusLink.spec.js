import React from 'react';
import { render, cleanup } from '@testing-library/react';
import FocusLink from '../FocusLink';

describe('FocusLink', () => {
  afterEach(cleanup);

  it('should render as a Link with string to', () => {
    const { container } = render(
      <FocusLink className="demo-class" to="/demo/path">
        Click me
      </FocusLink>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as a Link with object to', () => {
    const { container } = render(
      <FocusLink
        className="demo-class"
        to={{ pathname: '/demo/path', search: '?id=1' }}
      >
        Click me
      </FocusLink>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as a Link and merge router state', () => {
    const { queryByText } = render(
      <FocusLink
        className="demo-class"
        to={{ pathname: '/demo/path', state: { someKey: 'someVal' } }}
      >
        Click me
      </FocusLink>
    );

    expect(
      queryByText('{"someKey":"someVal","focus":true}')
    ).toBeInTheDocument();
  });

  it('should render as a NavLink with string to', () => {
    const { container } = render(
      <FocusLink
        className="demo-class"
        to="/demo/path"
        activeClassName="active"
      >
        Click me
      </FocusLink>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as a NavLink with object to', () => {
    const { container } = render(
      <FocusLink
        className="demo-class"
        to={{ pathname: '/demo/path', search: '?id=1' }}
        activeClassName="active"
      >
        Click me
      </FocusLink>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as a NavLink and merge router state', () => {
    const { queryByText } = render(
      <FocusLink
        className="demo-class"
        activeClassName="active"
        to={{ pathname: '/demo/path', state: { someKey: 'someVal' } }}
      >
        Click me
      </FocusLink>
    );

    expect(
      queryByText('{"someKey":"someVal","focus":true}')
    ).toBeInTheDocument();
  });
});
