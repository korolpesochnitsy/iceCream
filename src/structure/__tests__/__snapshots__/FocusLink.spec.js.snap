// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusLink should render as a Link with object to 1`] = `
<a
  href="/demo/path?id=1"
>
  Click me
  <span>
    {"focus":true}
  </span>
  <span
    data-testid="linkType"
  >
    Link
  </span>
</a>
`;

exports[`FocusLink should render as a Link with string to 1`] = `
<a
  href="/demo/path"
>
  Click me
  <span>
    {"focus":true}
  </span>
  <span
    data-testid="linkType"
  >
    Link
  </span>
</a>
`;

exports[`FocusLink should render as a NavLink with object to 1`] = `
<a
  href="/demo/path?id=1"
>
  Click me
  <span>
    {"focus":true}
  </span>
  <span
    data-testid="linkType"
  >
    NavLink
  </span>
</a>
`;

exports[`FocusLink should render as a NavLink with string to 1`] = `
<a
  href="/demo/path"
>
  Click me
  <span>
    {"focus":true}
  </span>
  <span
    data-testid="linkType"
  >
    NavLink
  </span>
</a>
`;
